yarn-project/acir-simulator/src/client/unconstrained_execution.ts

```diff
@@ -54,6 +56,26 @@ export class UnconstrainedFunctionExecution {
       },
       getL1ToL2Message: ([msgKey]) => this.context.getL1ToL2Message(fromACVMField(msgKey)),
       getCommitment: ([commitment]) => this.context.getCommitment(this.contractAddress, commitment),
+      storageRead: async ([slot], [numberOfElements]) => {
+        if (aztecNode === undefined) {
+          this.log(`Aztec node is undefined, cannot read public storage`);
+          throw new Error(`Aztec node is undefined, cannot read public storage`);
+        }
+        const startStorageSlot = fromACVMField(slot);
+        const values = [];
+        for (let i = 0; i < Number(numberOfElements); i++) {
+          const storageSlot = startStorageSlot.value + BigInt(i);
+          const value = await aztecNode.getPublicStorageAt(this.contractAddress, storageSlot);
+          if (value === undefined) {
+            this.log(`Oracle storage read: slot=${storageSlot.toString()} value=undefined`);
+            throw new Error(`Oracle storage read: slot=${storageSlot.toString()} value=undefined`);
+          }
+          const frValue = Fr.fromBuffer(value);
```

Comment:
toACVMField can take in a Buffer as an arg. It would be cleaner to just store values as buffers, they can then be converted in one step at the end

Reply:
Mainly for consistency between the implementations in the executor and here. Also just made it super simple to handle the logging 🤷